src/irr/asset/CImageWriterOpenEXR.cpp

```diff
@@ -22,14 +22,91 @@ SOFTWARE.
 
 #ifdef _IRR_COMPILE_WITH_OPENEXR_WRITER_
 
+#include "openexr/IlmBase/Imath/ImathBox.h"
+#include "openexr/OpenEXR/IlmImf/ImfOutputFile.h"
+#include "openexr/OpenEXR/IlmImf/ImfChannelList.h"
+#include "openexr/OpenEXR/IlmImf/ImfChannelListAttribute.h"
+#include "openexr/OpenEXR/IlmImf/ImfStringAttribute.h"
+#include "openexr/OpenEXR/IlmImf/ImfMatrixAttribute.h"
+#include "openexr/OpenEXR/IlmImf/ImfArray.h"
+#include <algorithm>
+#include <iostream>
+#include <string>
+#include <unordered_map>
+
+#include "openexr/OpenEXR/IlmImf/ImfNamespace.h"
+namespace IMF = OPENEXR_IMF_NAMESPACE;
+namespace IMATH = IMATH_NAMESPACE;
+
 namespace irr
 {
 	namespace asset
 	{
+		using namespace IMF;
+		using namespace IMATH;
+
 		bool CImageWriterOpenEXR::writeAsset(io::IWriteFile* _file, const SAssetWriteParams& _params, IAssetWriterOverride* _override)
 		{
-			if (!_file)
-				return {};
+			if (!_override)
+				getDefaultOverride(_override);
+
+			SAssetWriteContext ctx{ _params, _file };
+
+			const asset::ICPUImage* image =
+			#ifndef _IRR_DEBUG
+				static_cast<const asset::ICPUImage*>(_params.rootAsset);
+			#else
+				dynamic_cast<const asset::ICPUImage*>(_params.rootAsset);
+			#endif
+			assert(image);
```

Comment:
we should proabably make a macro out of this or a template utility function.

Reply:
well, so where should it be put in files?

Reply:
probably a template static function in `IAsset`

Reply:
call it `castDown`

Reply:
shouldn't it be in `IAssetWriter`?
I get many errors because of not known type if it is put into `IAsset`